src/lib/expression/evaluation/like_matcher.hpp

```diff
@@ -46,6 +47,12 @@ class LikeMatcher {
    */
   static PatternTokens pattern_string_to_tokens(const pmr_string& pattern);
 
+  // Calculates the upper and lower bound of a given pattern. For example, with the pattern `Japan%`, the lower bound
+  // `Japan` and upper bound `Japao` is returned. The first value of the returned pair is the lower bound the second
+  // the upper bound. if the char ASCII value before the wildcard has the max ASCII value 127, the upper and lower bound
+  // are the same.
+  static std::pair<pmr_string, pmr_string> get_lower_upper_bound(const pmr_string& pattern);
+
```

Comment:
Mmh, shouldn't we handle leading wildcards here, too? Right now, those would be ignored, correct? Return an `optional<pair>`, which is `nullopt` for both leading wildcards and ASCII overflows?

Reply:
I think it is a good idea to use an `optional<pair>` for ASCII overflows. But if we have a leading wildcard we can still check, for example in the case of `LIKE`, if the lower bound is larger than the max value or smaller.

Reply:
Could you give an example?

Reply:
Sorry, my answer wasn't correct. I meant we can't handle both in this function. If, for example (in the case of `Like`), we have the pattern: `"aa<char>(127)%");` we would get `nullopt`. In this case, we actually still need the `lower_bound` to check if `max < lower_bound` applies. If max would be "aa", we could determine that the value is not contained.  If the function returns `nullopt` in the case of a leading wildcard, we could not distinguish between the ASCII Overflow and leading wildcards and could miss this case. I think the best solution would be to only return `nullopt ` if it is a leading wildcard.

Reply:
Mmh, maybe we should first list all conditions before we see how they can be represented. Personally, I don't care about the cases with an overflow not being optimized, tbh.

* regular case
* Leading wildcard
* ASCII overflow
* No wildcard

Anything missing?

Reply:
That's all the cases. If the case of an ASCII overflow doesn't need to be optimized we could use the solution where we can return `optional<pair>`. Then in the case of an ASCII overflow and Leading wildcard, we would `does_not_contain` would return false.

Reply:
> In the case of a leading wildcard does_not_contain needs to return true

I don't get this part. For a leading wildcard, we cannot give any guarantees about the data based on min/max, so why would we be able to guarantee that the statistics object does **not** contain the search value?

I tried to come up with an optional-based return value for lower/upper bounds. To me, this looks consistent - does it make any sense?

|                           | test%                       | %test   | test\x7F% | test                          | '' (empty string)         |
|---------------------------|-----------------------------|---------|-----------|-------------------------------|---------------------------|
| LikeMatcher::bounds()     | {test, tesu}                | nullopt | nullopt   | {test, test\0}                | {'', '\0'}                |
| does_not_contain(Like)    | max < test \|\| min >= tesu | false   | false     | max < test \|\| min >= test\0 | max < '' \|\| min >= '\0' |
| does_not_contain(NotLike) | min >= test && max < tesu   | false   | false     | min >= test && max < test\0   | min >= '\0' && max < '\0' |

Reply:
Also pinging @Bouncner to get some more input on this.

Reply:
Ok, that makes sense. I will adjust the code and add documentation.